### ISSUE TITLE
WarpSync.initializeUniverse reentrancy bug for minting REP.

### ISSUE THREAD

**Issue by MicahZoltu** [AugurProject/augur#5123]
TL;DR: `mint` calls into user code and thus you can execute a reentrancy attack against WarpSync's `awardRep` function, with an added bonus of repeated `createMarket`'s RepBond minting.

1. After universe creation the user calls `WarpSync.initializeUniverse`
1. This will check that the provided universe is known.
1. --> This will check the universe does not yet have a WarpSync market.
1. It will then call into `WarpSync.awardRep`
1. Before updating any state, that will call into `WarpSync.awardRep`
1. Before updating any state, that will call into `ReputationToken.mintForWarpSync`
1. Before updating any state, that will call into `ReputationToken.mint`
1. Before updating any state, that will call into `ReputationToken._mint`
1. `ReputationToken.supply` will be increased.
1. `ReputationToken.balances[tx.origin]` will be incresed.
1. `ReputationToken._callTokensReceived` will be called.
1. --> If `tx.origin` has registered an ERC1820 contract, it will be called.  This gives the caller an opportunity to call `WarpSync.initializeUniverse()` again.
1. During unwind, `ReputationToken._callTokensReceived` doesn't do anything else.
1. During unwind, `ReputationToken._mint` only fires events.
1. During unwind, `ReputationToken.mint` only fires events.
1. During unwind, `ReputationToken.mintForWarpSync` will call `Universe.updateForkValues`.
1. `Universe.updateForkValues` only executes assignments, it doesn't do any checks other than `isForking`.
1. During unwind, `WarpSync.awardRep` doesn't do anything else.
1. During unwind, `WarpSync.initializeUniverse` will call `WarpSync.createMarket`.
1. --> `WarpSync.createMarket` will finally make the update necessary to prevent `WarpSync.initializeUniverse` from being called again (but too late for reentrancy protection).
1. `WarpSync.createMarket` only does idempotent assignments with the exception of another call to `ReputationToken.mintForWarpSync`, which will mint excessive bonds for the next market (amplifying the attack if the attacker can repeat multiple days in a row).

**Comment by MicahZoltu** [comment]
Recommended fix: call `WarpSync.createMarket` prior to calling `WarpSync.awardRep`.
